src/api.js

```diff
@@ -683,8 +786,18 @@ API.prototype.declare = function(options, handler) {
       throw new Error('query.' + key + ' must be a RegExp or a function!');
     }
   });
+  assert(!options.deferAuth,
+    'deferAuth is deprecated! https://github.com/taskcluster/taskcluster-lib-api#request-handlers');
   if ('scopes' in options) {
-    scopes.validateScopeSets(options.scopes);
+    assert(scopes.validExpression(_.cloneDeepWith(options.scopes, scope => {
+      // This makes these template constructs valid parts of an expression
+      if (_.isObject(scope) && scope.for && scope.in && scope.each) {
+        return 'looping-template-construct';
```

Comment:
Perhaps better `['looping', 'template', 'construct']` since it will get turned into an array?

Reply:
It doesn't actually get turned into an array unfortunately. It merges its resulting array into the array it is a part of.